rotkehlchen/utils/network.py

```diff
@@ -16,6 +17,21 @@
 log = RotkehlchenLogsAdapter(logger)
 
 
+def create_session() -> requests.Session:
+    """Create a requests session with the configuration to retry a maximum
+    of 3 times on connection errors or DNS resolution errors.
+    """
+    session = requests.Session()
+    # from the requests docs about max_retries:
+    # The maximum number of retries each connection should attempt. Note, this applies only
+    # to failed DNS lookups, socket connections and connection timeouts, never to requests
+    # where data has made it to the server.
+    adapter = HTTPAdapter(max_retries=3)
```

Comment:
We do the same but with more details and more args set in premium/premium.py

I think the extra arguments are needed.

```python
        self.session = requests.session()
        # Make sure to have 3 retries on read/connect/other errors for all requests
        # The reason for this is that we have noticed that in unstable/slow connections
        # rotki.com server will close/cause the connection to result to a read timeout
        # At the moment this only happens for the backup upload endpoint. More info:
        # https://github.com/rotki/rotki/pull/6423
        adapter = requests.adapters.HTTPAdapter()
        # Have to set retries like this, since this granularity is not given by HTTPAdapter ctor
        adapter.max_retries = Retry(
            total=3,  # have to set each one individually as the code checks them all
            read=3,
            connect=3,
            other=3,
            allowed_methods=False,  # retry on all method verbs
        )
        self.session.mount('https://', adapter)
```

And it should also be abstracted together with that code and combined.

Reply:
Since we couldn't find references for why to set the different counts individually in the Retry object we agreed on going with `HTTPAdapter` and adjust later if needed. Also checked the code for possible sessions again and didn't find any